whipper/src/main/java/org/whipper/resultmode/CompareResultMode.java

```diff
@@ -65,16 +66,29 @@ public ResultHolder handleResult(Query q){
         ResultHolder out = new ResultHolder();
         OverrideFileSelector selector = new OverrideFileSelector(q.getScenario().getExpectedResultsDir());
         File result = selector.getExpectedResultFile(getExpectedResultFileName(q));
-        String expectedResultDirectoryName = result.getParentFile().getParentFile().getName();
-        try{
-            holder.buildResult(result, q);
-            boolean eq = holder.equals(q.getActualResult(), !q.getSql().toUpperCase().contains(" ORDER BY "), allowedDivergence);
-            if(!eq){
-                writeErrorFile(q, expectedResultDirectoryName);
-                out.setErrors(holder.getErrors());
+        if (result != null && result.exists()) {
+            String expectedResultDirectoryName = result.getParentFile().getParentFile().getName();
+            try{
+                holder.buildResult(result, q);
+                boolean eq = holder.equals(q.getActualResult(), !q.getSql().toUpperCase().contains(" ORDER BY "), allowedDivergence);
+                if(!eq){
+                    writeErrorFile(q, expectedResultDirectoryName);
+                    out.setErrors(holder.getErrors());
+                }
+            } catch (IOException ex){
+                out.setException(ex);
+            }
+
+        } else {
+            // this means that no file with expected result was found
+            try {
+                holder.buildFakeResultFromMissingFile("Unable to find expected result file for query", q);
+                out.setErrors(Collections.singletonList("Unable to find expected result file for query"));
```

Comment:
Same string. Use "out.setErrors(holder.getErrors());" just like you do on line 76.

Reply:
fixed